### DISCUSSION TITLE
Is there a difference?

### DISCUSSION THREAD

**Discussion by ShervanN** [ProfessionalCSharp/ProfessionalCSharp2021#188] [General]
Hello,

Two things:
1) I am not sure. Anyway, On page 542   ❘ CHAPTER 19 Networking, You wrote:

> To create an application hosting the Kestrel server, you can create an empty web application with the .NET CLI:
> dotnet new web.

dotnet new web command will create an asp.net core app with Top-level statements, while your descriptions (in a few lines later) are based without this feature.

2) On page 543,  When I execute dotnet new web in .NET 7, I have the below snippet in Program class,
```
namespace web2;

public class Program
{
    public static void Main(string[] args)
    {
        var builder = WebApplication.CreateBuilder(args);
        var app = builder.Build();

        app.MapGet("/", () => "Hello World!");

        app.Run();
    }
}

```
While snippet code in your book is:
```
using Microsoft.AspNetCore.Hosting;

public class Program
{
    public static void Main(string[] args)
    {
        CreateHostBuilder(args).Build().Run();
    }
    public static IHostBuilder CreateHostBuilder(string[] args) =>
    Host.CreateDefaultBuilder(args)
    .ConfigureWebHostDefaults(webBuilder =>
    {
        webBuilder.UseStartup<Startup>();
    });
}
```
Does the text of the book and the code in this section need to be updated?

**Comment by christiannagel** [comment]
.NET 6 added the `WebApplication` and `WebApplicationBuilder` classes which abstracts away the `Host` class and removes the requirement of the `Startup` class. 

The code in repo has been updated already. Now I also added information to the readme files:

https://github.com/ProfessionalCSharp/ProfessionalCSharp2021/blob/main/Dotnet6Updates.md

https://github.com/ProfessionalCSharp/ProfessionalCSharp2021/blob/main/2_Libs/Networking/Readme.md

Also check this:

[Upgrading an ASP.NET Core Web API Project to .NET 6 - Article](https://csharp.christiannagel.com/2022/02/22/upgrading-an-asp-net-core-web-api-project-to-net-6/)

[Upgrading an ASP.NET Core Web API Project to .NET 6 - Source Code](https://github.com/ProfessionalCSharp/ProfessionalCSharp2021/blob/main/5_More/Services/APIServiceUpdate)

**Reply by ShervanN** [reply]
Thank you, How about the first question?

**Reply by christiannagel** [reply]
I thought this was already included in the answer - but the first question was about the top-level statements?

Contrary to the opinion of reviewers I used top-level statements with most of the book samples when authoring the book. A reviewer complained the readers would be confused when the top-level statements are not created on creation of the application. The templates with .NET 6 changed to top-level statements - which I was expecting.

Now, most program templates create an initial application with top-level statements. With web applications before .NET 6, using top-level statements was not that big of an advantage because of the requirement of the `Startup` class. With .NET 6, the `WebApplication` and `WebApplicationBuilder` classes have been introduced. These classes simplify the startup code a lot by using top-level statements.

If you like, you can create a program without top-level statements with:

`dotnet new web --use-program-main`
